docusaurus/docs/tutorials/convert-a-frontend-datasource-to-backend.md

```diff
@@ -0,0 +1,419 @@
+	}
+
+	// Create slice of values for time and values.
+	times := make([]time.Time, len(body.DataPoints))
+	values := make([]float64, len(body.DataPoints))
+	for i, p := range body.DataPoints {
+		times[i] = p.Time
+		values[i] = p.Value
+	}
+
+	// Create frame and add it to the response
+	dataResp := backend.DataResponse{
+		Frames: []*data.Frame{
+			data.NewFrame(
+				"response",
+				data.NewField("time", nil, times),
+				data.NewField("values", nil, values),
+			),
+		},
+	}
+	return dataResp, err
+}
+```
+
+### Other resource requests
+
+Finally, there is an optional type of request that plugins may implement. This is what we call _resources_. Resources are additional endpoints that the plugin exposes and uses to populate the query editor or config editor. For example, you can use a resource to populate a dropdown menu with a list of available tables in a database.
+
+In a frontend data source, the plugin should define the resources in the `plugin.json` file as `routes` and use the `fetch` method to get the data. For example:
+
+```json title="src/plugin.json"
+{
+  "routes": [
+    {
+      "path": "tables",
+      "url": "https://api.example.com/api/v1/tables",
+      "method": "GET"
+    }
+  ]
+}
+```
+
+```typescript title="src/DataSource.ts"
+export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+  async getTables() {
+    const response = await lastValueFrom(
+      getBackendSrv().fetch<MetricsResponse>({
+        url: `${this.url}/tables`,
+        method: 'GET',
+      })
+    );
+    return response.data;
+  }
+}
+```
+
+:::note
+
+To keep things simple, authentication is omitted in this example.
+
+:::
+
+For a backend data source, the plugin should implement the `CallResourceHandler` interface. This interface should handle the different possible resources. For example:
+
+```go title="pkg/plugin/datasource.go"
+func NewDatasource(_ context.Context, _ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &Datasource{
+		CallResourceHandler: newResourceHandler(),
+	}, nil
+}
+
+func newResourceHandler() backend.CallResourceHandler {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/tables", handleTables)
+
+	return httpadapter.New(mux)
+}
+
+func handleTables(w http.ResponseWriter, r *http.Request) {
+  // Get tables
+  res, err :=	http.DefaultClient.Get("https://api.example.com/api/v1/tables")
+  // Handle errors (omited)
+	body, err := io.ReadAll(res.Body)
+  // Handle errors (omited)
+
+	w.Write(body)
+	w.WriteHeader(http.StatusOK)
+}
+```
+
+To request the resource in the frontend, you can use the methods exposed in the base class `DataSourceWithBackend` (for example, `getResource` or `postResource`):
+
+```typescript title="src/DataSource.ts"
+export class DataSource extends DataSourceWithBackend<MyQuery, MyDataSourceOptions> {
+  async getTables() {
+    const response = await this.getResource('tables');
+    return response;
+  }
+}
+```
+
```

Comment:
are there any trouble spots or commonly encountered problems or bugs that the plugin developer may face? if so, add a section to address them

Reply:
not that I know of